core/domain/html_cleaner.py

```diff
@@ -192,3 +193,401 @@ def get_rte_components(html_string: str) -> List[ComponentsDict]:
+        if 'raw_latex' not in math_content_list:
+            raise utils.ValidationError(
+                'Math tag does not have \'raw_latex-with-value\' '
+                'attribute.'
+            )
+
+        if math_content_list['raw_latex'].strip() in empty_values:
+            raise utils.ValidationError(
+                'Math tag \'raw_latex-with-value\' attribute should not '
+                'be empty.'
+            )
+
+        if 'svg_filename' not in math_content_list:
+            raise utils.ValidationError(
+                'Math tag does not have \'svg_filename-with-value\' '
+                'attribute.'
+            )
+
+        if math_content_list['svg_filename'].strip() in empty_values:
+            raise utils.ValidationError(
+                'Math tag \'svg_filename-with-value\' attribute should not '
+                'be empty.'
+            )
+
+        if math_content_list['svg_filename'].strip()[-4:] != '.svg':
+            raise utils.ValidationError(
+                'Math tag \'svg_filename-with-value\' attribute should '
+                'have svg extension.'
+            )
+
+    if is_tag_nested_inside_tabs_or_collapsible:
+        tabs_tags = soup.find_all('oppia-noninteractive-tabs')
+        if len(tabs_tags) > 0:
+            raise utils.ValidationError(
+                'Tabs tag should not be present inside another '
+                'Tabs or Collapsible tag.'
+            )
+
+        collapsible_tags = soup.find_all('oppia-noninteractive-collapsible')
+        if len(collapsible_tags) > 0:
+            raise utils.ValidationError(
+                'Collapsible tag should not be present inside another '
+                'Tabs or Collapsible tag.'
+            )
+
+
+def validate_tabs_and_collapsible_rte_tags(html_data: str) -> None:
+    """Validates `Tabs` and `Collapsible` RTE tags
+
+    Args:
+        html_data: str. The RTE content of the state.
+
+    Raises:
+        ValidationError. No tabs present inside the tab_contents attribute.
+        ValidationError. No title present inside the tab_contents attribute.
+        ValidationError. Title inside the tag is empty.
+        ValidationError. No content present inside the tab_contents attribute.
+        ValidationError. Content inside the tag is empty.
+        ValidationError. No content attributes present inside the tabs tag.
+        ValidationError. No collapsible content is present inside the tag.
+        ValidationError. Collapsible content-with-value attribute is not
+            present.
+        ValidationError. Collapsible heading-with-value attribute is not
+            present.
+        ValidationError. Collapsible heading-with-value attribute is empty.
+    """
+    soup, empty_values = _initialize_bs4_and_empty_values_variable(html_data)
+    tabs_tags = soup.find_all('oppia-noninteractive-tabs')
+    for tag in tabs_tags:
+        if tag.has_attr('tab_contents-with-value'):
```

Comment:
again do `if not: raise` instead of `else: raise`.

Reply:
Done